crates/rattler_solve/src/libsolv/wrapper/solver.rs

```diff
@@ -96,17 +89,11 @@ impl SolverRef {
     }
 
     /// Creates a transaction from the solutions found by the solver.
-    pub fn create_transaction(&mut self) -> Transaction {
-        let transaction =
-            NonNull::new(unsafe { ffi::solver_create_transaction(self.as_ptr().as_mut()) })
-                .expect("solver_create_transaction returned a nullptr");
-        Transaction::new(transaction)
-    }
-}
+    pub fn create_transaction(&self) -> Transaction {
+        let transaction = NonNull::new(unsafe { ffi::solver_create_transaction(self.raw_ptr()) })
```

Comment:
I think that this might be a legitimate error. This could happen if `solve` hasn't been called yet. Maybe turn this into a Result?

Reply:
In that case I think we should automatically create the transaction after solving, to remove altogether the possibility of misusing the API. I'll merge this into `Solver::solve`